packages/frontend-v2/components/AddCourseModal/AddCourseModal.tsx

```diff
@@ -41,30 +40,16 @@ export const AddCourseModal: React.FC<AddCourseModalProps> = ({
   isCourseInCurrTerm,
   addClassesToCurrTerm,
 }) => {
-  const [searchTerm, setSearchTerm] = useState("");
-  const [isLoading, setIsLoading] = useState(false);
-  const [searchResults, setSearchResults] = useState<ScheduleCourse2<null>[]>(
-    []
-  );
+  const [searchQuery, setSearchQuery] = useState("");
   const [selectedCourses, setSelectedCourses] = useState<
     ScheduleCourse2<null>[]
   >([]);
 
-  const router = useRouter();
-
-  const searchCourses = async () => {
-    if (searchTerm !== "") {
-      setIsLoading(true);
-      try {
-        const results = await SearchAPI.searchCourses(searchTerm);
-        setSearchResults(results);
-      } catch (error) {
-        // TODO: This an error from Search, we may want to handle this differently.
-        handleApiClientError(error as Error, router);
-      }
-      setIsLoading(false);
-    }
-  };
+  const {
+    courses,
+    isLoading: isCoursesLoading,
+    error,
+  } = useSearchCourses(searchQuery);
```

Comment:
Ahh didn't realize you could simply just pass in the state into the hook without having to do anything we discussed haha. Good call!!